server/src/main/java/org/candlepin/resource/EnvironmentResource.java

```diff
@@ -323,6 +328,29 @@ public JobDetail demoteContent(
         return detail;
     }
 
+    /**
+     * To make promotion transactional
+     * @param contentToPromote
+     * @param env
+     * @param contentIds
+     */
+    @Transactional
+    public void batchCreate(List<org.candlepin.model.dto.EnvironmentContent> contentToPromote,
```

Comment:
We could have this method return the collection rather than accept it as input to be modified.